onnxruntime/core/session/inference_session.cc

```diff
@@ -1990,22 +2023,25 @@ common::Status InferenceSession::ValidateOutputs(gsl::span<const std::string> ou
     return common::Status(common::ONNXRUNTIME, common::INVALID_ARGUMENT, "At least one output should be requested.");
   }
 
-  if (!p_fetches->empty() && (output_names.size() != p_fetches->size())) {
-    std::ostringstream ostr;
-    ostr << "Output vector incorrectly sized: output_names.size(): " << output_names.size()
-         << "p_fetches->size(): " << p_fetches->size();
-    return common::Status(common::ONNXRUNTIME, common::INVALID_ARGUMENT, ostr.str());
+  const auto fetches = gsl::make_span(*p_fetches);
+
+  if (!fetches.empty() && (output_names.size() != fetches.size())) {
+    return ORT_MAKE_STATUS(ONNXRUNTIME, INVALID_ARGUMENT,
+                           "Output vector incorrectly sized: output_names.size(): ", output_names.size(),
+                           "fetches.size(): ", fetches.size());
```

Comment:
Not really. In case only  output names are supplied, then we validate those in ValidateOutputs. But if OrtValues are present, then validation happens inside ValidateInputOutputs, because we query metadata for the OrtValues by name. These are not the same.

Reply:
ah ok. why is `p_fetches` not allowed to be null, but allowed to be an empty vector?

Reply:
I would have guessed that because p_fetches is coming from the client side (python, C#, etc.) we need something to populate as the ownership semantics if we allocated p_fetches would be unclear and/or harder to manage.